app/services/doc_auth/error_generator.rb

```diff
@@ -312,29 +328,51 @@ def generate_doc_auth_errors(response_info)
       metrics_error = metrics_error_handler.handle(response_info)
       return metrics_error.to_h if metrics_error.present? && !metrics_error.empty?
 
-      # check selfie error
-      selfie_error_handler = SelfieErrorHandler.new
-      selfie_error = selfie_error_handler.handle(response_info)
+      doc_auth_result = response_info[:doc_auth_result]
+      processed_alerts = response_info[:processed_alerts]
+
+      doc_auth_result_passed = doc_auth_result == LexisNexis::ResultCodes::PASSED.name
+      doc_auth_passed_or_attn_with_barcode = doc_auth_result_passed ||
+                                             attention_with_barcode_result(
+                                               doc_auth_result,
+                                               processed_alerts,
+                                             )
+      doc_auth_error_count =
+        doc_auth_passed_or_attn_with_barcode ?
+                               0 : response_info[:alert_failure_count]
+      known_error_count = doc_auth_error_count - unknown_fail_count
+
+      doc_auth_error_handler = DocAuthErrorHandler.new
+      doc_auth_error = doc_auth_error_handler.handle(response_info, known_error_count)
+
+      if doc_auth_error.present? && !doc_auth_error.empty?
+        return doc_auth_error.to_h
+      end
 
-      # if selfie itself is ok, but we have selfie related error
-      if selfie_error_handler.is_generic_selfie_error?(selfie_error)
-        return selfie_error_handler.selfie_general_failure_error
+      # check selfie error
+      if doc_auth_error_count < 1 && doc_auth_passed_or_attn_with_barcode
```

Comment:
The conditions for this if statement seem like they can be simpler since doc_auth_error_count is already looking at the `doc_auth_passed_or_attn_with_barcode` field?

Reply:
It does look that way, but when I try removing it, the following test in `document_capture_spec` fails:

`when doc auth result is "attention" and face match errors`

It is confusing and sort of an edge case, but without this, a result that has:

```
doc_auth_result: attention (_not_ barcode)
face_match_result: fail
```

will show a selfie error. Keeping that check there prevents the selfie error from being shown and lets the attention error trickle down to the unknown error.

I was hoping it would be as easy as just deleting it. Unless there's an easier way to fix it I'm missing, maybe this is another case of "this code is not great, but it's functional and we're going to tear it out soon anyway"?

If I remove the previous use of `doc_auth_passed_or_attn_with_barcode` in `doc_auth_error_count`, a lot more tests fail. I think because without it in the logic, `response_info[:alert_failure_count]` will be more than 1, and then it will pass the check on the line you highlighted, meaning it will skip selfie errors when that's what we want to show.

Reply:
Ok, I think you're right that this is another place where the code is confusing right now and we should change it as little as possible to get things working.